persistence-jdbc/core/src/main/scala/com/lightbend/lagom/internal/persistence/jdbc/SlickProvider.scala

```diff
@@ -41,30 +42,38 @@ private[lagom] class SlickProvider(
 
   val autoCreateTables: Boolean = createTables.getBoolean("auto")
 
-  private val asyncExecConfig =
-    new AsyncExecutorConfig(system.settings.config.getConfig("jdbc-defaults.slick.async-executor"))
-
   if (dbApi != null) {
-    // Work around https://github.com/playframework/playframework/issues/7262
-    // Set the system property
-    System.setProperty(Context.PROVIDER_URL, "/")
 
-    // the data source as configured by Play
-    val dataSource = dbApi.database("default").dataSource
+    val namingContext = new InitialContext()
 
-    // the slick db configured with an async executor
-    val slickDb =
-      SlickDbProvider(
-        dataSource,
-        system.settings.config.getConfig("jdbc-defaults.slick.async-executor")
-      )
+    dbApi.databases().foreach { playDb =>
 
-    val namingContext = new InitialContext()
-    // bind the DB configured with an AsyncExecutor
-    namingContext.rebind("DefaultDB", slickDb)
+      val dbName = playDb.name
+
+      // each configured DB having an async-executor section
+      // is entitled to for Slick DB configured and bounded to a JNDI name
+      for {
+        dbConfig <- Try(system.settings.config.getConfig(s"db.$dbName"))
+        asyncExecConfig <- Try(dbConfig.getConfig("async-executor"))
```

Comment:
Maybe we should expose the config from `Database` in Play so we can clean this up in a future version?

Reply:
Yes, I think so.